prometheus_juju_exporter/collector.py

```diff
@@ -0,0 +1,45 @@
+from prometheus_client import CollectorRegistry, Gauge
+
+from prometheus_juju_exporter.config import Config
+from prometheus_juju_exporter.logging import get_logger
+
+
+class CollectorDaemon:
+    """Core class of the cloudstats exporter daemon."""
+
+    def __init__(self, registry=None, debug=False):
+        """Create new collector daemon and configure runtime environment."""
+        self.config = Config().get_config()
+        self.logger = self.setup_logging(debug=debug)
+        self._registry = registry or CollectorRegistry()
+        self.metrics = {}
+
+    def setup_logging(self, debug):
```

Comment:
what's the benefit in wrapping get_logger in setup_logging?

Reply:
After re-assessment, I don't think there is added value using a separate function here. So I made the change accordingly